tutorials/sphinx-tutorials/multiagent_competitive_ddpg.py

```diff
@@ -0,0 +1,992 @@
+            .item()
+        )
+        episode_reward_mean_map[group].append(episode_reward_mean)
+
+    pbar.set_description(
+        ", ".join(
+            [
+                f"episode_reward_mean_{group} = {episode_reward_mean_map[group][-1]}"
+                for group in env.group_map.keys()
+            ]
+        ),
+        refresh=False,
+    )
+    pbar.update()
+
+######################################################################
+# Results
+# -------
+#
+# Let's plot the mean reward obtained per episode
+#
+# To make training last longer, increase the ``n_iters`` hyperparameter.
+#
+
+fig, axs = plt.subplots(2, 1)
+for i, group in enumerate(env.group_map.keys()):
+    axs[i].plot(episode_reward_mean_map[group], label=f"Episode reward mean {group}")
+    axs[i].set_ylabel("Reward")
+    axs[i].axvline(
+        x=iteration_when_stop_training_evaders,
+        label="Agent (evader) stop training",
+        color="orange",
+    )
+    axs[i].legend()
+axs[-1].set_xlabel("Training iterations")
+plt.show()
+
+######################################################################
+# Render
+# ------
+#
+# *Rendering instruction are for VMAS*, aka when running with ``use_vmas=True``.
+#
+# If you are running this in a machine with GUI, you can render the trained policy by running:
+#
+# .. code-block:: python
+#
+#    from torchrl.envs.utils import ExplorationType, set_exploration_type
+#    with torch.no_grad(), set_exploration_type(ExplorationType.MODE):
+#       env.rollout(
+#           max_steps=max_steps,
+#           policy=agents_exploration_policy,
+#           callback=lambda env, _: env.render(),
+#           auto_cast_to_device=True,
+#           break_when_any_done=False,
+#       )
+#
+# If you are running this in Google Colab, you can render the trained policy by running:
+#
+# .. code-block:: bash
+#
+#    !sudo apt-get update
+#    !sudo apt-get install python3-opengl xvfb
+#    !pip install pyvirtualdisplay
+#
+# .. code-block:: python
+#
+#    import pyvirtualdisplay
+#    display = pyvirtualdisplay.Display(visible=False, size=(1400, 900))
+#    display.start()
+#    from PIL import Image
+#    from torchrl.envs.utils import ExplorationType, set_exploration_type
+#
+#    def rendering_callback(env, td):
+#        env.frames.append(Image.fromarray(env.render(mode="rgb_array")))
+#    env.frames = []
+#    scenario_name = "simple_tag"
+#    with torch.no_grad(), set_exploration_type(ExplorationType.MODE):
+#       env.rollout(
+#           max_steps=max_steps,
+#           policy=agents_exploration_policy,
+#           callback=rendering_callback,
+#           auto_cast_to_device=True,
+#           break_when_any_done=False,
+#       )
+#    env.frames[0].save(
+#        f"{scenario_name}.gif",
+#        save_all=True,
+#        append_images=env.frames[1:],
+#       duration=3,
+#       loop=0,
+#    )
+#
+#    from IPython.display import Image
+#    Image(open(f"{scenario_name}.gif", "rb").read())
```

Comment:
For consistency with the rest of the lib I'd rather use VIdeoRecorder
https://github.com/pytorch/rl/pull/2070

Reply:
This is for interactive rendering, not for video recording. These envs have a GUI rendering function and do not return pixels in the data

Reply:
It is not trivial as the env is batched. To use the recorder I would have to call the render in the callback of the rollout and write it to the pixels key of the td. But i can't since the render is from a single env and the td has the batch_size containing data from multiple environments.

These are just a few lines to display the gif in colab and they are also commented out. 

To use a recorder here i would have to create a bunch of infrastructure and I think it would just get out of scope.

Reply:
I'll patch things for you it should be simple